src/scripts/active-inactive-orgunit/index.js

```diff
@@ -170,138 +167,158 @@ OrganisationUnitActivator.prototype.processOrgUnitsFromParentLevel = function (d
     });
 };
 
+/**
+ * Process all the pull organisationUntis responses
+@param error Contains the error
+@param response Contains the response
+@param body Contains the body
+ */
+OrganisationUnitActivator.prototype.processOrgUnitResponse = function (error, response, body, dataElement, dataValues) {
+    if (error != undefined) {
+        console.error("Error loading orgUnits",
+            error);
+        this.asyncCalls--;
+        return;
+    }
+
+    console.info("Response body: ", body);
+    var organisationUnits = JSON.parse(body).organisationUnits;
+    if (organisationUnits == undefined) {
+        console.error("Error: OrganisationUnits not Found.", "dataelement: " + dataelement.id);
+        this.asyncCalls--;
+        return;
+    }
+    console.info("Found " +
+        organisationUnits.length +
+        " dataElements \n\t" +
+        organisationUnits.map(function (organisationUnit) { return organisationUnit.id }).join("\n\t")
+    );
+
+    this.prepareDataValues(organisationUnits, dataElement, dataValues);
+    this.asyncCalls--;
+    this.pushDataValues(dataValues);
+}
 
-OrganisationUnitActivator.prototype.prepareAndPushDataValues = function (organisationUnits, dataElement) {
+/**
+ * Loops all the organisation units and prepare their dataSets
+@param organisationUnits Contains all the organisationUntis
+@param dataElement It is the organisationUnit dataelement for this loop.
+ */
+OrganisationUnitActivator.prototype.prepareDataValues = function (organisationUnits, dataElement, dataValues) { 
     console.log("\nPreparing dataValues");
     var _this = this;
     organisationUnits.forEach(function (organisationUnit) {
-        _this.prepareDataSet(organisationUnit, dataElement);
+        _this.prepareDataOrgUnitDataValues(organisationUnit, dataElement, dataValues);
     });
-    console.info("\nPending asyncalls: " + this.asyncCalls);
-    if (this.asyncCalls <= 0) {
-        console.log("Pushing dataValues");
-        this.pushDataValues(this.buildDataValues(this.dataValues), dataElement.dataElementGroup);
-        console.log("The dataValues created from the dataElementGroup " + dataElement.dataElementGroup + " was pushed");
-    }
 };
+
 /**
- * Prepares and pushes the dataSet.
- */
-OrganisationUnitActivator.prototype.prepareDataSet = function (orgUnit, dataElement) {
+ * Prepares the dataSet.*
+@param orgUnit Contains the organisation unit uid and the closed and opening dates used to set the orgUnit as active or inactive
+@param dataElement The dataelement to be pushed with this organisation Unit
+*/
+OrganisationUnitActivator.prototype.prepareDataOrgUnitDataValues = function (orgUnit, dataElement, dataValues) {
     //Parse the server dates.
     if (orgUnit.closedDate != undefined) {
-        orgUnit.closedDate = this.parseDateFromDhis(orgUnit.closedDate);
+        orgUnit.closedDate =  dateFormatter.parseDateFromDhis(orgUnit.closedDate);
```

Comment:
We can add this `if (taka != undefined)` inside the parseDateFromDhis so we make the method more robust and we remove code. A good old friend used to tell me that good coders remove lines of code while keeping the same functionality ;)